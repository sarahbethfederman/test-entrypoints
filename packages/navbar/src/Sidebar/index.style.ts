import styled, { css } from 'styled-components';
import { bg } from '@lendi-ui/color';
import { Spacing } from '@lendi-ui/spacing';
import { State, mixin } from '@lendi-ui/transition';
import { depth } from '@lendi-ui/depth';
import { MenuButton } from '../Header/index.style';
import * as ZINDEX from '../constants/z-index';

export interface WrapperProps {
  side: 'left' | 'right';
  transition: State;
}

export const Wrapper = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  width: 22.5rem;
  overflow: hidden;
  transition: transform 0.25s;
  ${bg('shade.0')} ${depth(1)}
  z-index: ${ZINDEX.SIDEBAR};

  ${({ side }: WrapperProps) => {
    if (side === 'left') {
      return css`
        left: 0;
        transform: translateX(-100%);
        ${CloseButton} {
          right: ${Spacing.xxxs};
        }
      `;
    } else {
      return css`
        right: 0;
        transform: translateX(100%);
        ${CloseButton} {
          left: ${Spacing.xxxs};
        }
      `;
    }
  }} ${({ side, transition }: WrapperProps) => {
  if (side === 'left') {
    return mixin(transition, {
      entering: { transform: 'translateX(0)' },
      entered: { transform: 'translateX(0)' },
      exit: { transform: 'translateX(0)' },
      exiting: { transform: 'translateX(-100%)' },
    });
  } else {
    return mixin(transition, {
      entering: { transform: 'translateX(0)' },
      entered: { transform: 'translateX(0)' },
      exit: { transform: 'translateX(0)' },
      exiting: { transform: 'translateX(100%)' },
    });
  }
}};
`;

export interface CloseButtonProps {
  side: 'left' | 'right';
}

export const CloseButton = styled(MenuButton)`
  position: absolute;
  top: ${Spacing.xxxs};
`;
