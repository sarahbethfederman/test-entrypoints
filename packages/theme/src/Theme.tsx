import * as React from 'react';
import styled, { ThemeProvider, createGlobalStyle, GlobalStyleComponent, DefaultTheme } from 'styled-components';
import { Breakpoint } from '@lendi-ui/breakpoint';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
const GlobalStyle: GlobalStyleComponent<{ theme: ThemeMap }, DefaultTheme> = createGlobalStyle`
  html, body {
    margin: 0;
    padding: 0;
    font-family: ${({ theme }) => theme.typography.body};
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: ${({ theme }) => theme.typography.heading};
    font-weight: 700;
  }

  input::-ms-clear {
    display: none;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: ${Breakpoint.md}) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors['shade.700']};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>
        {children}
        <GlobalStyle />
      </Root>
    </ThemeProvider>
  );
}
