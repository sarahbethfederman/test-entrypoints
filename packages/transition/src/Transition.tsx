import * as React from 'react';
import { polyfill } from 'react-lifecycles-compat';

export type State = 'enter' | 'entering' | 'entered' | 'exit' | 'exiting' | 'exited' | undefined;

export interface TransitionProps {
  isActive: boolean;
  timeout: number;
  isVisible?: boolean;
  shouldMountOnEnter?: boolean;
  shouldUnmountOnExit?: boolean;
  onChangeState?: (state: State) => void;
  children: (state?: State) => React.ReactNode;
}

export interface TransitionState {
  mounted: boolean;
  state?: State;
}
class Transition extends React.Component<TransitionProps, TransitionState> {
  public static getDerivedStateFromProps(
    props: TransitionProps,
    state: TransitionState
  ): Partial<TransitionState> | null {
    if (state.mounted === false) {
      if (props.isVisible && props.isActive) {
        return {
          mounted: true,
          state: 'enter',
        };
      }

      if (props.isVisible && !props.isActive) {
        return {
          mounted: true,
          state: 'exit',
        };
      }

      return {
        mounted: true,
      };
    } else {
      if (
        !props.isActive &&
        (state.state === undefined ||
          state.state === 'enter' ||
          state.state === 'entering' ||
          state.state === 'entered')
      ) {
        return {
          state: 'exit',
        };
      }

      if (
        props.isActive &&
        (state.state === undefined || state.state === 'exit' || state.state === 'exiting' || state.state === 'exited')
      ) {
        return {
          state: 'enter',
        };
      }
    }

    return null;
  }

  public state: TransitionState = {
    mounted: false,
  };

  // typed as `any` due to a conflict between lib DOM and @types/node typings
  private timeout: any = null;
  private raf: any = null;

  private notifyStateChange() {
    const { onChangeState } = this.props;
    const { state } = this.state;
    if (onChangeState) {
      onChangeState(state);
    }
  }

  private enter() {
    this.notifyStateChange();
    cancelAnimationFrame(this.raf);
    this.raf = requestAnimationFrame(() => {
      const { timeout } = this.props;
      clearTimeout(this.timeout);
      this.setState({ state: 'entering' }, () => {
        this.notifyStateChange();
        this.timeout = setTimeout(() => this.setState({ state: 'entered' }, () => this.notifyStateChange()), timeout);
      });
    });
  }

  private exit() {
    this.notifyStateChange();
    cancelAnimationFrame(this.raf);
    this.raf = requestAnimationFrame(() => {
      const { timeout } = this.props;
      clearTimeout(this.timeout);
      this.setState({ state: 'exiting' }, () => {
        this.notifyStateChange();
        this.timeout = setTimeout(() => this.setState({ state: 'exited' }, () => this.notifyStateChange()), timeout);
      });
    });
  }

  public componentDidMount() {
    if (this.state.state === 'enter') {
      setTimeout(() => this.enter(), 0);
    }
    if (this.state.state === 'exit') {
      this.exit();
    }
  }

  public componentDidUpdate(prevProps: TransitionProps, prevState: TransitionState) {
    if (prevState.state !== 'enter' && this.state.state === 'enter') {
      setTimeout(() => this.enter(), 0);
    }
    if (prevState.state !== 'exit' && this.state.state === 'exit') {
      this.exit();
    }
  }

  public componentWillUnmount() {
    cancelAnimationFrame(this.raf);
    clearTimeout(this.timeout);
  }

  public render() {
    const { shouldMountOnEnter, shouldUnmountOnExit, children } = this.props;
    const { state } = this.state;

    if (shouldMountOnEnter && state === undefined) {
      return null;
    }

    if (shouldUnmountOnExit && state === 'exited') {
      return null;
    }

    return children(state);
  }
}

polyfill(Transition);
export default Transition;

export interface MixinOptions {
  enter?: any;
  entering?: any;
  entered?: any;
  exit?: any;
  exiting?: any;
  exited?: any;
}

export const mixin = (state: State, options: MixinOptions) => {
  switch (state) {
    case 'enter':
    case 'entering':
    case 'entered':
    case 'exit':
    case 'exiting':
    case 'exited':
      return options[state];
    default:
      return undefined;
  }
};

export interface StyleOptions {
  enter?: { [name: string]: string };
  entering?: { [name: string]: string };
  entered?: { [name: string]: string };
  exit?: { [name: string]: string };
  exiting?: { [name: string]: string };
  exited?: { [name: string]: string };
}

export const style = (state: State, options: StyleOptions) => {
  switch (state) {
    case 'enter':
    case 'entering':
    case 'entered':
    case 'exit':
    case 'exiting':
    case 'exited':
      return options[state];
    default:
      return undefined;
  }
};
