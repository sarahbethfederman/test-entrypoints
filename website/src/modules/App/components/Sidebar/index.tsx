import * as React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

import Logo from '@lendi-ui/logo';
import Heading from '@lendi-ui/heading';
import { fg } from '@lendi-ui/color';

import metadata, { Doc } from '../../../../utils/info';
import { Wrapper, LogoWrapper, NavGroup, SubNav } from './index.style';

const foundations = ['color', 'typography', 'breakpoint', 'spacing', 'grid', 'reset', 'theme'];

const ignoreIndexDoc = (doc: Doc) => doc.name !== 'index';

const SidebarLink = styled(Link)`
  ${fg('primary.500')};
  text-decoration: none;
`;

const KickerLink = styled(Link)`
  ${fg('shade.500')};
  text-decoration: none;
  text-transform: uppercase;
  font-size: 12px;
`;

export class Sidebar extends React.Component {
  get foundations() {
    const { workspaces } = metadata;
    return workspaces
      .filter((workspace) => foundations.includes(workspace.name))
      .filter((workspace) => workspace.name !== 'reset' && workspace.name !== 'theme');
  }

  get components() {
    const { workspaces } = metadata;
    return workspaces.filter((workspace) => !foundations.includes(workspace.name));
  }

  render() {
    return (
      <Wrapper>
        <LogoWrapper>
          <SidebarLink to="/">
            <Logo />
          </SidebarLink>
        </LogoWrapper>

        <NavGroup>
          <SidebarLink to={`/overview`}>Overview</SidebarLink>
        </NavGroup>

        <NavGroup>
          <Heading size="sm">Getting Started</Heading>
          <SidebarLink to={`/getting-started/setup`}>Setup</SidebarLink>
        </NavGroup>

        <NavGroup>
          <Heading size="sm">Foundations</Heading>
          {this.foundations.map((workspace) => (
            <div key={workspace.name}>
              <KickerLink to={`/package/${workspace.name}`}>{workspace.name}</KickerLink>
              {Boolean(workspace.docs.filter(ignoreIndexDoc).length) && (
                <SubNav>
                  {workspace.docs.filter(ignoreIndexDoc).map((doc) => (
                    <div key={doc.name}>
                      <SidebarLink to={`/package/${workspace.name}/${doc.name}`}>{doc.name}</SidebarLink>
                    </div>
                  ))}
                </SubNav>
              )}
            </div>
          ))}
        </NavGroup>

        <NavGroup>
          <Heading size="sm">Components</Heading>
          {this.components.map((workspace) => (
            <div key={workspace.name}>
              <SidebarLink to={`/package/${workspace.name}`}>{workspace.name}</SidebarLink>
              {Boolean(workspace.docs.filter(ignoreIndexDoc).length) && (
                <SubNav>
                  {workspace.docs.filter(ignoreIndexDoc).map((doc) => (
                    <div key={doc.name}>
                      <SidebarLink to={`/package/${workspace.name}/${doc.name}`}>{doc.name}</SidebarLink>
                    </div>
                  ))}
                </SubNav>
              )}
            </div>
          ))}
        </NavGroup>
      </Wrapper>
    );
  }
}
