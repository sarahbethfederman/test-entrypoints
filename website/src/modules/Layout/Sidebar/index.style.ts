import styled from 'styled-components';
import { bg, color } from '@lendi-ui/color';
import { px, ml, mb } from '@lendi-ui/spacing';
import { body } from '@lendi-ui/typography';
import { gte } from '@lendi-ui/breakpoint';

export const Wrapper = styled.div`
  box-sizing: border-box;
  ${px({ mobile: 'xs', tablet: 'md' })}
  ${body()}
  ${bg('shade.0')}
  ${gte('mobile')`
    overflow: hidden;
  `}
  ${gte('tablet')`
    height: 100vh;
    overflow: auto;
  `}
`;

export const LogoWrapper = styled.div`
  display: flex;
  align-items: center;
  height: 60px;
  ${gte('tablet')`
    height: 112px;
  `}
  ${mb('lg')}
  border-bottom: 1px solid ${color('shade.200')};
`;

export const NavGroup = styled.div`
  ${mb('xl')};
`;

export const SubNav = styled.div`
  ${ml('sm')} ${mb('xs')};
`;
