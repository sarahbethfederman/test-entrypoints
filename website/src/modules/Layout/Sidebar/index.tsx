import * as React from 'react';
import { Link } from 'react-router-dom';
import styled, { css } from 'styled-components';

import Logo from '@lendi-ui/logo';
import Dropdown from '@lendi-ui/dropdown';
import { Heading, overline, link } from '@lendi-ui/typography';
import { pl } from '@lendi-ui/spacing';
import { color } from '@lendi-ui/color';

import { ThemeType, AppContext } from '../../Common';
import metadata, { Doc } from '../../../utils/info';
import { foundations } from '../../../utils/constants';
import { Wrapper, LogoWrapper, NavGroup, SubNav } from './index.style';

const themeItems: { value: ThemeType; label: string }[] = [
  { value: 'Lendi', label: 'Lendi' },
  { value: 'Domain', label: 'Domain' },
];

const ignoreIndexDoc = (doc: Doc) => doc.name !== 'index';

const SidebarLink = styled(Link)`
  display: block;
  ${link({ color: 'primary.500' })} text-decoration: none;
`;

const SidebarHeader = styled(SidebarLink)`
  ${overline};
`;

const SidebarGroup = styled.div`
  ${pl('xxxs')} ${({ isSelected = false }: { isSelected: boolean }) => {
    if (isSelected) {
      return css`
        border-left: 4px solid ${color('secondary.500')};
      `;
    }
    return css`
      border-left: 4px solid transparent;
    `;
  }};
`;

export interface SidebarProps {
  match: string;
}

export class Sidebar extends React.Component<SidebarProps> {
  get foundations() {
    const { workspaces } = metadata;
    return workspaces.filter((workspace) => foundations.includes(workspace.name));
  }

  get components() {
    const { workspaces } = metadata;
    return workspaces.filter((workspace) => !foundations.includes(workspace.name));
  }

  render() {
    const { match } = this.props;

    return (
      <Wrapper>
        <LogoWrapper>
          <SidebarLink to="/">
            <Logo />
          </SidebarLink>
        </LogoWrapper>

        <NavGroup>
          <Heading size="sm">Getting Started</Heading>
          <SidebarLink to="/pages/getting-started">Setup</SidebarLink>
          <SidebarLink to="/pages/whitelist">Whitelist</SidebarLink>
          <SidebarLink to="/pages/changes">Recent Changes</SidebarLink>
        </NavGroup>

        <NavGroup>
          <Heading size="sm">Select theme</Heading>
          <AppContext.Consumer>
            {({ theme, changeTheme }) => (
              <Dropdown
                size={{ mobile: 'sm', tablet: 'md', desktop: 'lg' }}
                items={themeItems}
                isFullWidth
                value={theme}
                onChange={(value) => changeTheme(value as ThemeType)}
              />
            )}
          </AppContext.Consumer>
        </NavGroup>

        <NavGroup>
          <Heading size="sm">Foundations</Heading>
          {this.foundations.map((workspace) => (
            <SidebarGroup key={workspace.name} isSelected={workspace.name === match}>
              <SidebarHeader size="lg" to={`/packages/${workspace.name}`}>
                {workspace.name}
              </SidebarHeader>
              {Boolean(workspace.docs.filter(ignoreIndexDoc).length) && (
                <SubNav>
                  {workspace.docs.filter(ignoreIndexDoc).map((doc) => (
                    <div key={doc.name}>
                      <SidebarLink to={`/packages/${workspace.name}/${doc.slug}`}>{doc.name}</SidebarLink>
                    </div>
                  ))}
                </SubNav>
              )}
            </SidebarGroup>
          ))}
        </NavGroup>

        <NavGroup>
          <Heading size="sm">Components</Heading>
          {this.components.map((workspace) => (
            <SidebarGroup key={workspace.name} isSelected={workspace.name === match}>
              <SidebarHeader size="lg" to={`/packages/${workspace.name}`}>
                {workspace.name}
              </SidebarHeader>
              <SubNav>
                {Boolean(workspace.docs.filter(ignoreIndexDoc).length) && (
                  <>
                    {workspace.docs.filter(ignoreIndexDoc).map((doc) => (
                      <div key={doc.name}>
                        <SidebarLink to={`/packages/${workspace.name}/${doc.slug}`}>{doc.name}</SidebarLink>
                      </div>
                    ))}
                  </>
                )}
              </SubNav>
            </SidebarGroup>
          ))}
        </NavGroup>
      </Wrapper>
    );
  }
}
