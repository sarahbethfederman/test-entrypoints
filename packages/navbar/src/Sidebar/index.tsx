import * as React from 'react';
import * as Fragment from 'react-dot-fragment';
import Overlay from '@lendi-ui/overlay';
import Transition, { State } from '@lendi-ui/transition';
import { Wrapper, CloseButton, OverlayWrapper } from './index.style';
import { Close } from '@lendi-ui/icon';

export interface SidebarProps {
  side: 'left' | 'right';
  show: boolean;
  onHide?: () => void;
  children?: React.ReactNode;
}

class Sidebar extends React.Component<SidebarProps> {
  render() {
    const { side, show, onHide, children } = this.props;
    return (
      <Fragment>
        <OverlayWrapper>
          <Overlay show={show} onHide={onHide} />
        </OverlayWrapper>
        <Transition active={show} timeout={250}>
          {(state: State) => (
            <Wrapper side={side} transition={state}>
              <CloseButton onClick={onHide} aria-label="close">
                <Close color="shade.300" />
              </CloseButton>
              {children}
            </Wrapper>
          )}
        </Transition>
      </Fragment>
    );
  }
}

export default Sidebar;
