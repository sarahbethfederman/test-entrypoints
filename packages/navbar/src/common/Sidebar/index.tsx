import React, { Fragment } from 'react';
import Overlay from '@lendi-ui/overlay';
import Transition, { State } from '@lendi-ui/transition';
import { Wrapper, CloseButton, OverlayWrapper } from './index.style';
import { Close } from '@lendi-ui/icon';
import { AnalyticsContext } from '@lendi-ui/utils';
import { WindowPosition } from '@lendi/lendi-analytics-package';

export interface SidebarProps {
  side: 'left' | 'right';
  show: boolean;
  onHide?: () => void;
  children?: React.ReactNode;
  includeClose?: boolean;
}

class Sidebar extends React.Component<SidebarProps> {
  static contextType: any = AnalyticsContext;
  render() {
    const { side, show, includeClose = true, onHide = () => {}, children } = this.props;
    return (
      <Fragment>
        <OverlayWrapper>
          <Overlay isVisible={show} onHide={onHide} />
        </OverlayWrapper>
        <Transition isActive={show} timeout={250}>
          {(state: State) => (
            <Wrapper side={side} transition={state}>
              {includeClose && (
                <CloseButton
                  onClick={() => {
                    this.context.analyticsForNavigation('icon', WindowPosition.navigation_left);
                    onHide();
                  }}
                  aria-label="close"
                >
                  <Close color="shade.300" />
                </CloseButton>
              )}
              {children}
            </Wrapper>
          )}
        </Transition>
      </Fragment>
    );
  }
}

export default Sidebar;
