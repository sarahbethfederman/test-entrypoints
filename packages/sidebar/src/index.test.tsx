import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import Theme from '@lendi-ui/theme';
import Overlay from '@lendi-ui/overlay';
import Transition from '@lendi-ui/transition';

import { Wrapper, SidebarContent, SidebarFooter } from './index.style';
import Sidebar, { SidebarProps } from './index';

let element: ReactWrapper<SidebarProps>;

const render = ({ ...props }: SidebarProps) => {
  element = mount(
    <Theme>
      <Sidebar isVisible={true} {...props} />
    </Theme>
  );
};

const renderWithChildren = ({ ...props }: SidebarProps) => {
  element = mount(
    <Theme>
      <Sidebar isVisible={true} {...props}>
        <Sidebar.Content>content</Sidebar.Content>
        <Sidebar.Footer>footer</Sidebar.Footer>
      </Sidebar>
    </Theme>
  );
};

describe('Sidebar', () => {
  beforeEach(() => {
    render({});
  });

  it('should render <Sidebar />', () => {
    expect(element.find(Sidebar)).toHaveLength(1);
    expect(element.find(Sidebar)).toMatchSnapshot();
  });

  it('should render <Overlay />', () => {
    expect(element.find(Overlay)).toHaveLength(1);
  });

  it('should render <Transition />', () => {
    expect(element.find(Transition)).toHaveLength(2);
  });

  it('should render <Wrapper />', () => {
    expect(element.find(Wrapper)).toHaveLength(1);
  });

  it("should render <Wrapper /> with style 'left:0'", () => {
    expect(element.find(Wrapper)).toHaveStyleRule('left', '0');
  });
});

describe('Sidebar with children', () => {
  beforeEach(() => {
    renderWithChildren({});
  });

  it('should render <SidebarContent />', () => {
    expect(element.find(SidebarContent)).toHaveLength(1);
  });

  it('should render <SidebarFooter />', () => {
    expect(element.find(SidebarFooter)).toHaveLength(1);
  });
});
