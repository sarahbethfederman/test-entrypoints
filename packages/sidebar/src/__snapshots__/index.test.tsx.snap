// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render <Sidebar /> 1`] = `
.c0 {
  position: fixed;
  z-index: 2;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0,0,0,1);
  opacity: 0.4;
}

.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  bottom: 0;
  width: calc(100% - 3rem);
  max-width: 22.5rem;
  overflow: hidden;
  -webkit-transition: -webkit-transform 0.25s;
  -webkit-transition: transform 0.25s;
  transition: transform 0.25s;
  z-index: 999;
  background-color: rgba(255,255,255,1);
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.1), 0 2px 2px 0 rgba(0,0,0,0.12), 0 1px 3px 0 rgba(0,0,0,0.14);
  left: 0;
  top: 0px;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

<Sidebar
  isVisible={true}
>
  <Overlay
    isVisible={true}
    zIndex={2}
  >
    <div>
      <Fade
        isActive={true}
        isVisible={true}
        shouldMountOnEnter={true}
        shouldUnmountOnExit={true}
      >
        <Transition
          isActive={true}
          isVisible={true}
          shouldMountOnEnter={true}
          shouldUnmountOnExit={true}
          timeout={250}
        >
          <div
            style={
              Object {
                "opacity": "0",
                "transition": "opacity 0.25s",
              }
            }
          >
            <styled.div
              onClick={[Function]}
              zIndex={2}
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "c0",
                      "rules": Array [
                        "
  position: fixed;
  z-index: ",
                        [Function],
                        ";
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: ",
                        [Function],
                        ";
  opacity: 0.4;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                onClick={[Function]}
                zIndex={2}
              >
                <div
                  className="c0"
                  onClick={[Function]}
                />
              </StyledComponent>
            </styled.div>
          </div>
        </Transition>
      </Fade>
    </div>
  </Overlay>
  <Transition
    isActive={true}
    timeout={250}
  >
    <styled.div
      side="left"
      top={0}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": false,
              "lastClassName": "c1",
              "rules": Array [
                "
  ",
                "
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  letter-spacing: normal;
",
                "
  display: flex;
  flex-direction: column;
  position: fixed;
  bottom: 0;
  width: calc(100% - ",
                "3rem",
                ");
  max-width: 22.5rem;
  overflow: hidden;
  transition: transform 0.25s;
  z-index: ",
                "999",
                ";
  ",
                "
          background-color: ",
                [Function],
                ";
        ",
                " ",
                "box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.1),
        0 2px 2px 0 rgba(0, 0, 0, 0.12),
        0 1px 3px 0 rgba(0, 0, 0, 0.14);",
                ";

  ",
                [Function],
                " ",
                [Function],
                ";
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        side="left"
        top={0}
      >
        <div
          className="c1"
        />
      </StyledComponent>
    </styled.div>
  </Transition>
</Sidebar>
`;
